tests/test_onnx/test_head.py

```diff
@@ -224,3 +216,292 @@ def test_yolov3_head_get_bboxes():
+
+    onnx_outputs = verify_model(cls_scores + bboxes + centerness)
+
+    torch_outputs = wrap_model.forward(cls_scores, bboxes, centerness)
+    torch_outputs = convert_result_list(torch_outputs)
+    torch_outputs = [
+        torch_output.detach().numpy() for torch_output in torch_outputs
+    ]
+
+    # match torch_outputs and onnx_outputs
+    for i in range(len(onnx_outputs)):
+        np.testing.assert_allclose(
+            torch_outputs[i], onnx_outputs[i], rtol=1e-03, atol=1e-05)
+
+
+def fsaf_config():
+    """FSAF Head Config."""
+    cfg = dict(
+        anchor_generator=dict(
+            type='AnchorGenerator',
+            octave_base_scale=1,
+            scales_per_octave=1,
+            ratios=[1.0],
+            strides=[8, 16, 32, 64, 128]))
+
+    test_cfg = mmcv.Config(
+        dict(
+            deploy_nms_pre=0,
+            min_bbox_size=0,
+            score_thr=0.05,
+            nms=dict(type='nms', iou_threshold=0.5),
+            max_per_img=100))
+
+    model = FSAFHead(num_classes=4, in_channels=1, test_cfg=test_cfg, **cfg)
+    model.eval()
+    model.requires_grad_(False)
+    return model
+
+
+def test_fsaf_head_forward_single():
+    """Test RetinaNet Head forward_single() in torch and onnxruntime env."""
+    fsaf_model = fsaf_config()
+
+    feat = torch.rand(1, fsaf_model.in_channels, 32, 32)
+    ort_validate(fsaf_model.forward_single, feat)
+
+
+def test_fsaf_head_forward():
+    """Test RetinaNet Head forward in torch and onnxruntime env."""
+    fsaf_model = fsaf_config()
+
+    s = 128
+    feats = [
+        torch.rand(1, fsaf_model.in_channels, s // (2**(i + 2)),
+                   s // (2**(i + 2)))
+        for i in range(len(fsaf_model.anchor_generator.strides))
+    ]
+    ort_validate(fsaf_model.forward, feats)
+
+
+def test_fsaf_head_get_bboxes():
+    """Test RetinaNet Head get_bboxes in torch and onnxruntime env."""
+    fsaf_model = fsaf_config()
+    s = 256
+    img_metas = [{
+        'img_shape_for_onnx': (s, s, 3),
+        'scale_factor': 1,
+        'pad_shape': (s, s, 3),
+        'img_shape': (s, s, 2)
+    }]
+
+    # The data of fsaf_head_get_bboxes.pkl contains two parts:
+    # cls_score(list(Tensor)) and bboxes(list(Tensor)),
+    # where each torch.Tensor is generated by torch.rand().
+    # the cls_score's size: (1, 4, 64, 64), (1, 4, 32, 32),
+    # (1, 4, 16, 16), (1, 4, 8, 8), (1, 4, 4, 4).
+    # the bboxes's size: (1, 4, 64, 64), (1, 4, 32, 32),
+    # (1, 4, 16, 16), (1, 4, 8, 8), (1, 4, 4, 4).
+    fsaf_head_data = 'fsaf_head_get_bboxes.pkl'
+    feats = mmcv.load(osp.join(data_path, fsaf_head_data))
+    cls_score = feats[:5]
+    bboxes = feats[5:]
+
+    fsaf_model.get_bboxes = partial(
+        fsaf_model.get_bboxes, img_metas=img_metas, with_nms=False)
+    wrap_model = WrapFunction(fsaf_model.get_bboxes)
+    wrap_model.cpu().eval()
+    with torch.no_grad():
+        torch.onnx.export(
+            wrap_model, (cls_score, bboxes),
+            'tmp.onnx',
```

Comment:
At the end of function `ort_validate`, if file `onnx_io` exists, it should be removed.